indico/htdocs/js/indico/jquery/permissions_widget.js

```diff
@@ -0,0 +1,342 @@
+            var $permissionsList = $('<ul>').appendTo($permissions);
+            permissions.forEach(function(item) {
+                var permissionInfo = self.options.permissionsInfo[item];
+                $permissionsList.append(
+                    $('<li>', {class: 'i-label bold ' + permissionInfo.css_class, title: permissionInfo.description})
+                        .append(permissionInfo.title)
+                );
+            });
+            if (principal._type !== 'DefaultEntry') {
+                $permissions.append(this._renderPermissionsButtons(principal, permissions));
+            }
+            return $permissions;
+        },
+        _renderPermissionsButtons: function(principal, permissions) {
+            var $buttonsGroup = $('<div>', {class: 'group flexrow'});
+            $buttonsGroup.append(this._renderEditBtn(principal, permissions), this._renderDeleteBtn(principal));
+            return $buttonsGroup;
+        },
+        _renderEditBtn: function(principal, permissions) {
+            if (principal._type === 'IPNetworkGroup') {
+                return $('<button>', {
+                    type: 'button',
+                    class: 'i-button text-color borderless icon-only icon-edit disabled',
+                    title: $T.gettext('IP networks cannot have management permissions')
+                });
+            } else {
+                return $('<button>', {
+                    'type': 'button',
+                    'class': 'i-button text-color borderless icon-only icon-edit',
+                    'data-href': build_url(Indico.Urls.EventPermissions, {confId: this.options.eventId}),
+                    'data-title': $T.gettext('Assign Permissions'),
+                    'data-method': 'POST',
+                    'data-ajax-dialog': '',
+                    'data-params': JSON.stringify({principal: JSON.stringify(principal), permissions: permissions})
+                });
+            }
+        },
+        _renderDeleteBtn: function(principal) {
+            var self = this;
+            return $('<button>', {
+                'type': 'button',
+                'class': 'i-button text-color borderless icon-only icon-remove',
+                'data-principal': JSON.stringify(principal)
+            }).on('click', function() {
+                var $this = $(this);
+                var title = $T.gettext("Delete entry '{0}'".format(principal.name || principal.id));
+                if (principal._type === 'Avatar' && principal.id === self.options.userId) {
+                    var message = $T.gettext("Are you sure you want to remove yourself from the list?");
+                    confirmPrompt(message, title).then(function() {
+                        self._updateItem($this.data('principal'), []);
+                    });
+                } else {
+                    self._updateItem($this.data('principal'), []);
```

Comment:
You could avoid having this line twice using something like this (copied from the old place where we had this check):

                if (currentUserId === userId) {
                    confirmed = confirmPrompt($T.gettext('Are you sure you want to remove yourself from the list?'),
                                              $T.gettext('Confirm action'));
                } else {
                    confirmed = $.Deferred().resolve();
                }

                cnonfirmed.then(...)